sklearn/meson.build

```diff
@@ -0,0 +1,140 @@
+fs = import('fs')
+
+cython_args = []
+
+# Platform detection
+is_windows = host_machine.system() == 'windows'
+is_mingw = is_windows and cc.get_id() == 'gcc'
+
+if is_mingw
+  # For mingw-w64, link statically against the UCRT.
+  gcc_link_args = ['-lucrt', '-static']
+  add_project_link_arguments(gcc_link_args, language: ['c', 'cpp'])
+  # Force gcc to float64 long doubles for compatibility with MSVC
+  # builds, for C only.
+  add_project_arguments('-mlong-double-64', language: 'c')
+endif
```

Comment:
Do we ever build scikit-learn with mingw on our CI? Has someone tried that it actually works?

I am a bit worried of including untested build config. At the same time, if those are the only lines needed to support `mingw` then that might be helpful to keep them for some users who don't want to rely on MSVC for some reason.

Adding a new CI config just to test those lines is probably to much extra maintenance cost for very users.